### DISCUSSION TITLE
how do I connect sqlitebrowser to VS C# project?

### DISCUSSION THREAD

**Discussion by amanda-futterman** [sqlitebrowser/sqlitebrowser#3360] [Q&A]
trying to connect my DB browser database to a C# project. LMK if you know how to.

**Comment by chrisjlocke** [comment]
I know how to for VB.Net, so the process should be the same for C#.

**Download the DLLs**
![](https://i.imgur.com/M69fw2d.png)

Head to [system.data.sqlite.org/](https://system.data.sqlite.org/index.html/doc/trunk/www/downloads.wiki) and download the 32-bit and 64-bit packages ( (1) and (2) )
Extract system.data.sqlite.dll from the 32-bit .zip and stick in the same directory as your .exe.
Extract sqlite.interop.dll from the 32-bit .zip and stick in the 'x86' directory in the directory where your .exe is.
Extract sqlite.interop.dll from the 64-bit .zip and stick in the 'x64' directory in the directory where your .exe is.

```
<bin>\App.exe (your application)
<bin>\App.dll (optional, managed-only application library assembly)
<bin>\System.Data.SQLite.dll (required, managed-only core assembly)
<bin>\x86\SQLite.Interop.dll (required, x86 native interop assembly)
<bin>\x64\SQLite.Interop.dll (required, x64 native interop assembly)
```

Reference SQLite.Interop.dll in Visual Studio.

In your app, create a sqliteconnection, pointing it to your database file.
open the connection.  Bask in the glory that the database is open.  All hail great database.  She lives!!!
Create a sqlCommand, giving it some SQL and pointing it to the connection.
Create a dataset and an adapter.  You can just create a reader, but this method allows throwing it to datagrids.
point the data adapter to the command, fill the adapter with the dataset
Nibble on the table object which is the result from your SQL .. either one row or a bunch of rows.

Yell back if you need an example, or if none of this makes sense.
Again, I can only provide vb examples, but converting from vb to c# is quite easy.

**Comment by scottfurry** [comment]
@amanda-futterman - A source I've come to trust over the years is "zetcode". The admin there has maintained numerous "first steps" pages for how to utilize some software library. Think of it as a "Hello World" for software libraries.

SQLite w/ C# is no exception - https://zetcode.com/csharp/sqlite/. Page walks through what you need to know to get going. Hopefully, this will get you started in a good direction.

After that, `Google Fu` what you need to know. It's all about asking good questions to get good answers.

**Reply by chrisjlocke** [reply]
Code Pilot and chatGPT is now the way.  Google?  Pah - ye olde technology. ;)

**Reply by scottfurry** [reply]
@chrisjlocke - Yes, I am an anachronistic dinosaur fossil. And damn proud if it. I've earned my grey hair honestly.

LFMF, young padawan.
